### PR TITLE
Delete button functionality

### PR DESCRIPTION
**Pull request by ErlingHauan**
## Description
When pressing the delete button on the dashboard, we now have a modal that makes the user confirm that they want to delete the form. If confirmed, HTTP Delete requests are sent to `/api/forms/:formId` (deletes the form) and `/api/submissions/form/:formId` (deletes submissions belonging to the form).

Also added a React Context, `FormContext`, because of excessive prop drilling in the dashboard.

## Screenshot
image

### CODE REVIEW

File: backend/src/Repositories/SubmissionRepository.cs [hunk 1]
```diff
@@ -11,7 +11,8 @@ public interface ISubmissionRepository
     Task<SubmissionEntity?> GetSingle(Guid submissionId);
     Task<List<SubmissionEntity>> GetFormSubmissions(Guid formId);
     Task<SubmissionEntity?> Create(SubmissionEntity entity);
-    Task<SubmissionEntity?> Delete(Guid submussionId);
+    Task<SubmissionEntity?> DeleteSingle(Guid submussionId);
```
Comment (JamalAlabdullah) [inline backend/src/Repositories/SubmissionRepository.cs:14]: typo in `submussionId` 🙂

File: frontend/main-app/src/components/DashboardAccordion/DashboardAccordion.tsx [hunk 2]
```diff
@@ -13,20 +13,21 @@ import axios from "axios";
...
+import { FormContext } from "../Dashboard";
+import { t } from "i18next";
 interface ButtonGroupProps {
   submissions: Submission[];
-  formTitle: string;
   formUrl: string;
 }
 
-const ButtonGroup: React.FC<ButtonGroupProps> = ({ submissions, formTitle, formUrl }) => {
+const ButtonGroup: React.FC<ButtonGroupProps> = ({ submissions, formUrl }) => {
```
Comment (JamalAlabdullah) [inline frontend/main-app/src/components/DashboardAccordion/DashboardAccordion.tsx:24]: I think we can refactur `DashboardAccordion `by moving `ButtonGroup `to a seperate component. I see that you did not build `ButtonGroup `in this PR , so maybe we can add a seperate  quilty code issue  and fix it later. 😊

File: frontend/main-app/src/components/DeleteForm/DeleteForm.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    if (cleanFormData.confirmDelete.toString().toLowerCase() === "delete") {
+      deleteForm(form.id.toString());
+    } else {
+      setFormError(t("dashboard.delete.error"));
+    }
+  };
+
+  return (
+    <form className={className} onSubmit={handleDelete}>
+      <Paragraph>{t("dashboard.delete.areYouSure")}</Paragraph>
```
Comment (JamalAlabdullah) [inline frontend/main-app/src/components/DeleteForm/DeleteForm.tsx:48]: What if we update this to "`dashboard.delete.form.confirmation`"🙂

File: frontend/main-app/src/components/DeleteForm/DeleteForm.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+import { getApiUrl } from "../Login/LoginUtils";
+import axios from "axios";
+import { Trans } from "react-i18next";
+import { t } from "i18next";
+
+interface ModalContentProps {
+  className: string;
+}
+
+const ModalContent: React.FC<ModalContentProps> = ({ className }) => {
```
Comment (JamalAlabdullah) [inline frontend/main-app/src/components/DeleteForm/DeleteForm.tsx:15]: I think this component can be refactured in a seperat issue later , by moving ModalContent to an own componenrt.

**Review by JamalAlabdullah — CHANGES_REQUESTED** [review]
Looks nice, and i tested it. 🙂

File: frontend/main-app/src/components/DashboardAccordion/DashboardAccordion.tsx (hunk 2 above)
Comment (ErlingHauan) [reply frontend/main-app/src/components/DashboardAccordion/DashboardAccordion.tsx:24]: Agreed, I did that in PR #164 😊

File: frontend/main-app/src/components/DeleteForm/DeleteForm.tsx (hunk 3 above)
Comment (ErlingHauan) [reply frontend/main-app/src/components/DeleteForm/DeleteForm.tsx:48]: Here we have a text divided into two paragraphs.
What about this?

```suggestion
      <Paragraph>{t("dashboard.delete.form.confirm1")}</Paragraph>
      <Paragraph>
        <Trans i18nKey="dashboard.delete.form.confirm2" />
      </Paragraph>
```

```
"dashboard.delete.form.confirm1": "Are you sure you want to delete this form? Both the form and its submissions will be deleted permanently.",
"dashboard.delete.form.confirm2": "To confirm, type <strong>delete</strong> in the box below.",
  
```

Comment (JamalAlabdullah) [reply frontend/main-app/src/components/DeleteForm/DeleteForm.tsx:48]: Maybe the numbers  will not be descriptive enough. 😊, what about ;
```
dashboard.delete.form.confirm
dashboard.delete.form.instructions..confirm
```

Comment (ErlingHauan) [reply frontend/main-app/src/components/DeleteForm/DeleteForm.tsx]: Good idea!
I landed on this:
```
dashboard.delete.form.confirm.question
dashboard.delete.form.confirm.instruction
```

File: frontend/main-app/src/components/DeleteForm/DeleteForm.tsx (hunk 4 above)
Comment (ErlingHauan) [reply frontend/main-app/src/components/DeleteForm/DeleteForm.tsx]: I did a quick refactor just now - can you check it out in the latest commits?
image

**Review by JamalAlabdullah — DISMISSED** [review]
Veldig bra 🥇

**Pull request merged**